events/ri/src/intTest/java/org/projectnessie/events/ri/kafka/ITKafkaEventSubscriber.java

```diff
@@ -0,0 +1,566 @@
+          .withExposedPorts(8081)
+          .withEnv("SCHEMA_REGISTRY_HOST_NAME", "schemaregistry")
+          .withEnv("SCHEMA_REGISTRY_KAFKASTORE_BOOTSTRAP_SERVERS", "broker:9092");
+
+  private static final String TOPIC_NAME = "nessie.events";
+
+  private static final Reference BRANCH =
+      ImmutableReference.builder()
+          .simpleName("branch1")
+          .fullName("refs/heads/branch1")
+          .type("BRANCH")
+          .build();
+  private static Properties consumerConfig;
+
+  @BeforeAll
+  static void setUpConfig() throws IOException {
+    consumerConfig = generateConsumerConfig();
+    Properties producerConfig = generateProducerConfig();
+    Path file = writeConfig(producerConfig);
+    System.setProperty(
+        KafkaEventSubscriber.NESSIE_KAFKA_PROPERTIES_FILE_SYS_PROP,
+        file.toAbsolutePath().toString());
+  }
+
+  @Test
+  public void testProduceConsume() {
+    try (KafkaEventSubscriber producer = loadKafkaEventSubscriberSpi();
+        KafkaConsumer<String, Object> consumer = new KafkaConsumer<>(consumerConfig)) {
+
+      producer.onSubscribe(createSubscription());
+      consumer.subscribe(List.of(TOPIC_NAME));
+
+      emitEvents(producer);
+      assertEventsReceived(consumer);
+    }
+  }
+
+  private static Properties generateProducerConfig() {
+    Properties props = new Properties();
+    props.put(KafkaEventSubscriber.ConfigOption.TOPIC_NAME.getKey(), TOPIC_NAME);
+    props.put(KafkaEventSubscriber.ConfigOption.REPOSITORY_IDS.getKey(), "repo1, repo2");
+    props.put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, StringSerializer.class);
+    props.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, KafkaAvroSerializer.class);
+    props.put(ProducerConfig.ACKS_CONFIG, "0");
+    props.put(ProducerConfig.RETRIES_CONFIG, 0);
+    props.put(ProducerConfig.BATCH_SIZE_CONFIG, 8);
+    addCommonConfig(props);
+    return props;
+  }
+
+  private static Properties generateConsumerConfig() {
+    Properties props = new Properties();
+    props.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class);
+    props.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, KafkaAvroDeserializer.class);
+    props.put(ConsumerConfig.GROUP_ID_CONFIG, "test");
+    props.put(ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG, true);
+    props.put(ConsumerConfig.AUTO_COMMIT_INTERVAL_MS_CONFIG, 1000);
+    props.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest");
+    props.put(KafkaAvroDeserializerConfig.SPECIFIC_AVRO_READER_CONFIG, true);
+    addCommonConfig(props);
+    return props;
+  }
+
+  private static void addCommonConfig(Properties props) {
+    props.put(CommonClientConfigs.BOOTSTRAP_SERVERS_CONFIG, KAFKA.getBootstrapServers());
+    props.put(
+        AbstractKafkaSchemaSerDeConfig.SCHEMA_REGISTRY_URL_CONFIG,
+        "http://localhost:" + SCHEMA_REGISTRY.getMappedPort(8081));
+    props.put(
+        AbstractKafkaSchemaSerDeConfig.VALUE_SUBJECT_NAME_STRATEGY, TopicRecordNameStrategy.class);
```

Comment:
Where do we specify what part (which field) of the record is used by the `TopicRecordNameStrategy`?

Reply:
In this strategy, the schema is recorded under the subject `<topic name>-<schema name>`, so you don't need to select a field in the record. E.g. for the Avro `ReferenceEvent` message, the schema name is `ReferenceEvent`.